controllers/v1/library/categories.js

```diff
@@ -237,21 +288,191 @@ module.exports = class LibraryCategories extends Abstract {
+				result: result.data,
+			}
+		} catch (error) {
+			return {
+				status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+				message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+				errorObject: error,
+			}
+		}
+	}
+
+	/**
+	 * @api {patch} /project/v1/library/categories/move/:id
+	 * @apiVersion 1.0.0
+	 * @apiName move
+	 * @apiGroup LibraryCategories
+	 * @apiHeader {String} X-auth-token Authenticity token
+	 * @apiUse successBody
+	 * @apiUse errorBody
+	 */
+	async leaves(req) {
+		try {
+			const result = await libraryCategoriesHelper.getLeaves(req)
+			return {
+				success: true,
+				message: result.message,
+				result: result.data,
+			}
+		} catch (error) {
+			return {
+				status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+				message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+				errorObject: error,
+			}
+		}
+	}
+
+	/**
+	 * @api {post} /project/v1/library/categories/bulk
+	 * @apiVersion 1.0.0
+	 * @apiName bulk
+	 * @apiGroup LibraryCategories
+	 * @apiHeader {String} X-auth-token Authenticity token
+	 * @apiUse successBody
+	 * @apiUse errorBody
+	 */
+	async bulk(req) {
+		try {
+			const categories = req.body.categories || []
+			const tenantId = req.body.tenantId || req.userDetails.tenantAndOrgInfo.tenantId
+			const orgId = req.body.orgId || req.userDetails.tenantAndOrgInfo.orgId[0]
+
+			const result = await libraryCategoriesHelper.bulkCreate(categories, tenantId, orgId, req.userDetails)
+			return {
+				success: true,
+				message: result.message,
+				result: result.data,
+			}
+		} catch (error) {
+			return {
+				status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+				message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+				errorObject: error,
+			}
+		}
+	}
+
+	/**
+	 * @api {delete} /project/v1/library/categories/delete/:id
+	 * @apiVersion 1.0.0
+	 * @apiName delete
+	 * @apiGroup LibraryCategories
+	 * @apiHeader {String} X-auth-token Authenticity token
+	 * @apiUse successBody
+	 * @apiUse errorBody
+	 */
+	async delete(req) {
+		try {
+			const categoryId = req.params._id
+			const tenantId = req.query.tenantId || req.userDetails.tenantAndOrgInfo.tenantId
+			const orgId = req.query.orgId || req.userDetails.tenantAndOrgInfo.orgId[0]
+
+			const result = await libraryCategoriesHelper.delete(categoryId, tenantId, orgId)
+			if (result.success) {
+				return {
+					success: true,
+					message: result.message,
+					result: result.data,
+				}
+			} else {
+				throw {
+					message: result.message,
+					status: result.status || HTTP_STATUS_CODE.bad_request.status,
+				}
+			}
+		} catch (error) {
+			return {
+				status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+				message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+				errorObject: error,
+			}
+		}
+	}
+
+	/**
+	 * @api {get} /project/v1/library/categories/details/:id
```

Comment:
@Sachintechjoomla is this /project/v1/library/categories/details/:id or /project/v1/library/categories/:id

Reply:
project/v1/library/categories/details/:id this is previously 
But as standard i removed details word in url so now 
/project/v1/library/categories/id